frontend/src/composables/use-dialog-stack.ts

```diff
@@ -0,0 +1,29 @@
+import { computed, ref } from "vue"
+
+const stack = ref<string[]>([])
+
+export const useDialogStack = () => {
+  const push = (id: string) => {
+    stack.value.push(id)
+  }
+  const pop = () => {
+    stack.value.pop()
+  }
+  const clear = () => {
+    stack.value = []
+  }
+  const indexOf = (id: string) => {
+    return stack.value.indexOf(id)
+  }
+
+  const activeDialog = computed(() => stack.value[stack.value.length - 1])
+
+  return {
+    stack,
+    push,
+    pop,
+    clear,
+    indexOf,
+    activeDialog,
+  }
+}
```

Comment:
Can we document what activeDialog means in terms of the actual UI here? For example saying it's the "parent dialog which other dialogs can appear on top of but has no dialogs below it" or something. That's a _terrible_ example but hopefully conveys what I mean.